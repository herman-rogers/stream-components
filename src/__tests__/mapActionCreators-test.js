import '@babel/polyfill';
import { mapActionCreators } from '../';
import rx5Adapter from '../rx5Adapter';
import rxConnect, { getAdapter } from '../rxConnect';

describe('mapActionCreators RxJS 5', () => {
  let Rx;

  beforeEach(() => {
    rxConnect.adapter = rx5Adapter;
    Rx = getAdapter().Rx;
  });

  it('passes non-observables as is', async () => {
    const props = await mapActionCreators({ a: 123, b: 'hi!' }).toPromise();

    expect(props).toMatchSnapshot();
  });

  it('strips dollar sign from Observable property names', async () => {
    const actions = {
      a$: new Rx.Subject()
    };

    const props = await mapActionCreators(actions).toPromise();

    expect(props).toMatchSnapshot();
  });

  it('creates FuncSubject-like action', async () => {
    const actions = {
      a$: new Rx.BehaviorSubject()
    };

    const props = await mapActionCreators(actions).toPromise();

    props.a(1, 2, 3);

    expect(actions.a$.getValue()).toMatchSnapshot();
  });
});
