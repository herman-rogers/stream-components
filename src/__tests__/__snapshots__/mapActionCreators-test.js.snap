// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapActionCreators RxJS 5 creates FuncSubject-like action 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`mapActionCreators RxJS 5 passes non-observables as is 1`] = `
Object {
  "a": 123,
  "b": "hi!",
}
`;

exports[`mapActionCreators RxJS 5 strips dollar sign from Observable property names 1`] = `
Object {
  "a": [Function],
}
`;
